src/DI/Container.php

```diff
@@ -108,4 +111,17 @@ public function tabs() {
 	public function ui() {
 		return new UIServices($this);
 	}
+
+    /**
+     * Get interface to the http service.
+     *
+     * @return GlobalHttpState
+     */
+	public function http() {
+	    return new HTTPServices(
```

Comment:
Hi @klees 

Yes or at least I hope so. You are absolutely right.
That is the first time I used a DIC which can't resolve the dependencies by it self, that confused me a bit.
Is that the proper way of doing that? (ilInitialisation)

```
        $container['http.request_factory'] = function ($c) {
            return new \ILIAS\HTTP\Request\RequestFactoryImpl();
        };

        $container['http.response_factory'] = function ($c) {
            return new \ILIAS\HTTP\Response\ResponseFactoryImpl();
        };

        $container['http.cookie_jar_factory'] = function ($c) {
            return new \ILIAS\HTTP\Cookies\CookieJarFactoryImpl();
        };

        $container['http.request_sender_strategy'] = function ($c) {
            return new \ILIAS\HTTP\Response\Sender\DefaultResponseSenderStrategy();
        };

        $container['http'] = function ($c) {
            return new \ILIAS\DI\HTTPServices(
                $c['http.request_sender_strategy'],
                $c['http.cookie_jar_factory'],
                $c['http.request_factory'],
                $c['http.response_factory']
            );
        };
```
I also moved the request and response out of the DIC into the GlobalHttpState.
The new global HTTPServices constructor would now look like this. (without the wrong indentation)

```
public function __construct(
        ResponseSenderStrategy $senderStrategy,
        CookieJarFactory $cookieJarFactory,
        RequestFactory $requestFactory,
        ResponseFactory $responseFactory
    )
	{
		$this->sender = $senderStrategy;
		$this->$cookieJarFactory = $cookieJarFactory;

		$this->requestFactory = $responseFactory;
		$this->responseFactory = $responseFactory;
	}
```
About the injection of the GlobalHttpState.
That is a nice idea, but does that work for all services? The Web Access Checker for example has an own entry point and is never initialized by ILIAS.

I will implement the changes if they are ok. Otherwise, I would be glad to discuss further actions via Skype.